packages/agreement-lifecycle/src/filters/attributesFilter.ts

```diff
@@ -19,7 +19,17 @@ export const filterVerifiedAttributes = (
       att.verifiedBy.find(
         (v) =>
           v.id === producerId &&
-          (!v.extensionDate || v.extensionDate > new Date())
+          !att.revokedBy.find((revocation) => revocation.id === v.id) &&
+          ((!v.extensionDate && !v.expirationDate) ||
+            (v.extensionDate &&
+              v.expirationDate &&
+              v.extensionDate > new Date() &&
+              v.expirationDate > new Date()) ||
+            (v.extensionDate
+              ? v.extensionDate > new Date()
+              : v.expirationDate
+              ? v.expirationDate > new Date()
+              : false))
```

Comment:
As part of this PR I also decided to fix this logic used to filter attributes when adding them to the agreement, based on the discussion in this Slack thread: https://buildo.slack.com/archives/C0527LU1K4G/p1730297018249059